src/System.Text.Json/System/Text/Json/JsonWriter.cs

```diff
@@ -342,42 +321,36 @@ private void WriteStartAttribute(string name)
 
             WriteSpacing();
             WriteQuotedString(name);
-            WriteChar(JsonConstants.KeyValueSeperator);
+            WriteControl(JsonConstants.KeyValueSeperator);
 
             if (_prettyPrint)
-                WriteChar(JsonConstants.Space);
+                WriteControl(JsonConstants.Space);
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private void WriteChar(char value)
+        private void WriteControl(byte value)
         {
-            var output = _output;
             if (UseFastUtf8)
             {
-                if (output.Buffer.Length < 1)
-                    output.Enlarge();
-
-                output.Buffer.DangerousGetPinnableReference() = (byte)value;
-                output.Advance(1);
+                EnsureBuffer(1).DangerousGetPinnableReference() = value;
+                _output.Advance(1);
             }
             else if (UseFastUtf16)
             {
-                if (output.Buffer.Length < 2)
-                    output.Enlarge(2);
-
                 unsafe
                 {
-                    Unsafe.AsRef<char>(Unsafe.AsPointer<byte>(ref output.Buffer.DangerousGetPinnableReference())) = value;
+                    var buffer = EnsureBuffer(2);
+                    Unsafe.AsRef<char>(Unsafe.AsPointer<byte>(ref buffer.DangerousGetPinnableReference())) = (char)value;
```

Comment:
Yes, I can change this to a different construct. Unsafe.As<byte, char> also does not require unsafe block and is good for writing a single char.